### DISCUSSION TITLE
"Invalid file format" error when installing SPK file

### DISCUSSION THREAD

**Discussion by cardinal-tom** [runfalk/synology-wireguard#105] [General]
Hi, I downloaded the WireGuard-apollolake-1.0.20201221.spk file from the releases page for my DS218+, but when I try to install it I get the error "Invalid file format. Please contact the package developer". This is on DSM 7.0-41890. Any thoughts? I don't suppose it actually is invalid. I tried downloading it a second time just in case there was an issue with the first download, but no luck.

Tom

**Comment by gschmalz** [comment]
Just had the same problem for rtd1296 architecture. In my case I forgot to delete an older version of Wireguard before installing the new version.

**Reply by cardinal-tom** [reply]
I did get it to work in the end, after a lot of tidying up - had to log in via shell and go through the system partition to find some related build files. I'm not sure what actually got it working. Thanks for posting, and prompting my reply.

**Reply by aurelized** [reply]
hello, I am having the same issue with my DS1618+. Could you elaborate a bit more on what you did to get it to work, please?

**Reply by cardinal-tom** [reply]
It looks like I logged in via telnet then searched for Wireguard related folders

`sudo find . -iname '*ireguard*'`

found the directory where I'd installed Wireguard from and then deleted anything left over from the installation (shell scripts, an SPK, the directory build_artifacts) and tried again. Looking at the output from that there's also a /wireguard_compile and /wireguard_tarballs directory around too, delete them also! Let me know how it goes.

**Reply by aurelized** [reply]
it worked!!!

thanks a lot

**Reply by aurelized** [reply]
but then the package wont run :/

**Reply by cardinal-tom** [reply]
Have you tried wg-easy? [https://github.com/WeeJeWel/wg-easy](url) I'm planning on replacing my existing installation with this at some point.

**Reply by toddrob99** [reply]
@aurelized if you didn't uncheck the "run after install" box when you installed the spk, check the readme.

**Reply by aurelized** [reply]
> @aurelized if you didn't uncheck the "run after install" box when you installed the spk, check the readme.

you are right, i forgot about it, thank you

**Reply by aurelized** [reply]
> Have you tried wg-easy? [https://github.com/WeeJeWel/wg-easy](url) I'm planning on replacing my existing installation with this at some point.

i have not, but i will, thank you

**Comment by aurelized** [comment]
~$ WireGuard has been successfully started

thank you!
